src/shogun/lib/SGMatrix.h

```diff
@@ -309,6 +333,68 @@ template<class T> class SGMatrix : public SGReferencedData
 		/** fill matrix with zeros */
 		void zero();
 
+		/**
+		 * fill matrix with uniformly distributed randoms
+		 * Float Types: [0, 1)
+		 * Int Types: std::numeric_limit<T>::(min->max)
+		 * Rest: disabled
+		**/
+		// vectorized available
+		template <typename U>
+		using enable_simd_float = std::enable_if_t<is_any<U, float64_t>::value>;
```

Comment:
Cute!
I am wondering whether we should maybe move all the template magic out of SG* since that is supposed to only be a wrapper class. Maybe Random itself.... or linalg?

Reply:
I think it should stay here - Ive noticed that macros have been used at come places to replicate this functionality which imho is a bad practice. eg. [here](https://github.com/shogun-toolbox/shogun/blob/develop/src/shogun/lib/SGVector.cpp#L415). Keeping such examples would help push towards using SFINAE/tag dispatch for overload resolution.

Reply:
Totally right!
If you want to have a stab at doing some of this in a better way, feel free to submit a draft PR for discussing it! Fixing the equals would be a start

Reply:
Sure, I could look up and try to submit a few more once this is done.